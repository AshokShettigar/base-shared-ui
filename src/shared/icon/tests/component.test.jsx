import { shallow } from 'enzyme';

import Icon from '../component';

describe('Icon component', () => {
    const requiredProps = {
        glyph: 'calendar'
    };

    it('should render component with default props', () => {
        const wrapper = shallow(<Icon {...requiredProps} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with class size prop', () => {
        const wrapper = shallow(<Icon {...requiredProps} size="tiny" />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with onClick prop', () => {
        const wrapper = shallow(<Icon {...requiredProps} onClick={() => {}} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with controlled prop', () => {
        const wrapper = shallow(<Icon {...requiredProps} controlled />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with active prop', () => {
        const wrapper = shallow(<Icon {...requiredProps} active />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with disabled prop', () => {
        const wrapper = shallow(<Icon {...requiredProps} disabled />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with required prop', () => {
        const wrapper = shallow(<Icon {...requiredProps} required />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with class names props', () => {
        const wrapper = shallow(<Icon {...requiredProps} className="className" glyphClassName="glyphClassName" />);

        expect(wrapper).toMatchSnapshot();
    });

    it('shouldn\'t render component', () => {
        const wrapper = shallow(<Icon {...requiredProps} rendered={false} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render hidden component', () => {
        const wrapper = shallow(<Icon {...requiredProps} hidden />);

        expect(wrapper).toMatchSnapshot();
    });
});
