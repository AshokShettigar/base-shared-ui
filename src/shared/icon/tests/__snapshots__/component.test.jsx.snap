// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component should render component with active prop 1`] = `
<Icon
  active={true}
  controlled={true}
  disabled={false}
  glyph="calendar"
  hidden={false}
  size="medium"
/>
`;

exports[`Icon component should render component with class names props 1`] = `
<Icon
  active={false}
  className="className"
  controlled={true}
  disabled={false}
  glyph="calendar"
  glyphClassName="glyphClassName"
  hidden={false}
  size="medium"
/>
`;

exports[`Icon component should render component with class size prop 1`] = `
<Icon
  active={false}
  controlled={true}
  disabled={false}
  glyph="calendar"
  hidden={false}
  size="tiny"
/>
`;

exports[`Icon component should render component with controlled prop 1`] = `
<Icon
  active={false}
  controlled={true}
  disabled={false}
  glyph="calendar"
  hidden={false}
  size="medium"
/>
`;

exports[`Icon component should render component with default props 1`] = `
<Icon
  active={false}
  controlled={true}
  disabled={false}
  glyph="calendar"
  hidden={false}
  size="medium"
/>
`;

exports[`Icon component should render component with disabled prop 1`] = `
<Icon
  active={false}
  controlled={true}
  disabled={true}
  glyph="calendar"
  hidden={false}
  size="medium"
/>
`;

exports[`Icon component should render component with onClick prop 1`] = `
<Icon
  active={false}
  controlled={true}
  disabled={false}
  glyph="calendar"
  hidden={false}
  onClick={[Function]}
  size="medium"
/>
`;

exports[`Icon component should render component with required prop 1`] = `
<Icon
  active={false}
  controlled={true}
  disabled={false}
  glyph="calendar"
  hidden={false}
  required={true}
  size="medium"
/>
`;

exports[`Icon component should render hidden component 1`] = `
<div
  hidden={true}
>
  <Icon
    active={false}
    controlled={true}
    disabled={false}
    glyph="calendar"
    hidden={true}
    size="medium"
  />
</div>
`;

exports[`Icon component shouldn't render component 1`] = `null`;
